### PR TITLE
Fixes #34576 - Additional steps before deleting repo from CVs

### PR DESCRIPTION
**Pull request by sjha4**
#### What are the changes introduced in this pull request?
Force deletion of repo should have additional layers of security when deleting repos from cv versions.
This PR adds an additional checkbox on the modal to confirm delete, and a setting which can be turned off to disallow force deletions of repositories.
#### What are the testing steps for this pull request?

1. Create a repo.
2. Add it to CVs and publish a few versions of those.
3. Go to Repo, action dropdown > Delete
4. You should see the modal with a checkbox.
5. Check it. Delete button should get enabled. Don't click it yet! Close the modal.
6. Go to settings > content > Allow deleting repositories in published content views : Set to No
7. Go back to same repo and Actions > Delete should not be an option. Instead you should see a Cannot remove with a help text.
8. Go to settings > content > Allow deleting repositories in published content views : Set to Yes
9. Go back to repository.
10. Delete.

### CODE REVIEW

**Comment by ianballou** [comment]
I'm trying to think where a user permission might be better than a setting.  I'm imagining this:

A sysadmin doesn't want the people with repo delete permissions to be force deleting repos from content views.  However, they like to use the new feature because they know what they're doing.  So, they keep the setting off, and then go flip it whenever they want to force delete a repo.  There is a chance however that an unauthorized user could then force delete a repo because the setting is global. Plus, the sysadmin could forget to flip it back. So, it ends up being annoying for the sysadmin and potentially doesn't suit their authorization needs.

If there's a user permission that starts off as admin only, then it's sure that only the sysadmin can force delete repos.  And, if there are other trusted users, they could do it too.

**Comment by sjha4** [comment]
> I'm trying to think where a user permission might be better than a setting. I'm imagining this:
> 
> A sysadmin doesn't want the people with repo delete permissions to be force deleting repos from content views. However, they like to use the new feature because they know what they're doing. So, they keep the setting off, and then go flip it whenever they want to force delete a repo. There is a chance however that an unauthorized user could then force delete a repo because the setting is global. Plus, the sysadmin could forget to flip it back. So, it ends up being annoying for the sysadmin and potentially doesn't suit their authorization needs.
> 
> If there's a user permission that starts off as admin only, then it's sure that only the sysadmin can force delete repos. And, if there are other trusted users, they could do it too.

So I discussed the permissions vs settings approach with Sayan and Justin. Thought here is that delete_repo permission should be sufficient to manage permissions for deleting repos cause that is intuitive. A new permission would be need to be added to existing users who have already been granted delete_repo permissions. The setting is global and on by default making this feature immediately available for use without any actions on the part of the admin. Assumption is that admins and non-admins with delete permissions on repo will be alerted enough with the additional checkbox that needs a conscious decision on their part to delete it across CVs..

**TLDR: A new permission would be overly restrictive and require work to enable this feature. A setting which is on by default makes it immediately available for all users with delete permissions while providing flexibility to turn it off.**

**Comment by ianballou** [comment]
> So I discussed the permissions vs settings approach with Sayan and Justin. Thought here is that delete_repo permission should be sufficient to manage permissions for deleting repos cause that is intuitive. A new permission would need to be added to existing users who have already been granted delete_repo permissions. The setting is global and on by default making this feature immediately available for use without any actions on the part of the admin. Assumption is that admins and non-admins with delete permissions on repo will be alerted enough with the additional checkbox that needs a conscious decision on their part to delete it across CVs..
> 
> **TLDR: A new permission would be overly restrictive and require work to enable this feature. A setting which is on by default makes it immediately available for all users with delete permissions while providing flexibility to turn it off.**

I gotcha, so we're not digging too much into the aspect of limiting who can force delete repos. That's fine then since it's on by default.

File: engines/bastion/app/assets/javascripts/bastion/components/bst-modal.directive.js [hunk 1]
```diff
@@ -59,6 +59,7 @@ angular.module('Bastion.components').directive('bstModal',
 
                     modalInstance.result.then(function () {
                         scope.action();
+                    }, function() {
```
Comment (ianballou) [inline engines/bastion/app/assets/javascripts/bastion/components/bst-modal.directive.js:62]: Why'd you need to add this in?

Comment (sjha4) [reply engines/bastion/app/assets/javascripts/bastion/components/bst-modal.directive.js:62]: Uggh.. 😬  Ya, the bastion library somehow was suppressing a weird console warning on modal close. When I overrode the modal footer for repo deletion to add the customization with checkbox and button disabling, it started appearing. 
This is to suppress that error back. More details here: https://github.com/angular-ui/bootstrap/issues/6501#issuecomment-292881004

Comment (ianballou) [reply engines/bastion/app/assets/javascripts/bastion/components/bst-modal.directive.js:62]: Oh weird! Okay, makes sense then.
